### PR TITLE
Support gdb pretty-printing displayHint type "map" by processing map …

### PR DESCRIPTION
**Pull request by paulmaybee**
…children as name-value pairs.

### CODE REVIEW

File: src/MIDebugEngine/Engine.Impl/Variables.cs [hunk 1]
```diff
@@ -561,6 +573,21 @@ private Task FetchChildren()
                         i++;
                     }
                 }
+                else if (IsMapType())
+                {
+                    //
+                    // support for gdb's pretty-printing built-in displayHint "map"
+                    //
+                    List<VariableInformation> listChildren = new List<VariableInformation>();
+                    for(int p = 0; p+1 < children.Length; p+=2)
```
Comment (gregg-miskelly) [inline src/MIDebugEngine/Engine.Impl/Variables.cs]: Might want to add a comment explaining this format a bit, and why we are skipping two at a time.

**Comment by gregg-miskelly** [comment]
Signing off with one minor comment.

File: src/MIDebugEngine/Engine.Impl/Variables.cs (hunk 1 above)
Comment (jacdavis) [inline src/MIDebugEngine/Engine.Impl/Variables.cs]: Yeah, this is quite confusing for someone with no experience with these mi records.

**Comment by paulmaybee** [comment]
Comments added

**Pull request merged**